### PR TITLE
Actually use TargetOS=AnyOS in the build-test-job.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Match the name that's used in the job title and enable us to use TargetOS to pre-filter OS-specific tests with the new Xunit test wrapper generator.

### CODE REVIEW

File: eng/pipelines/common/templates/runtimes/build-test-job.yml [hunk 1]
```diff
@@ -147,7 +147,7 @@ jobs:
     # Zip and publish managed test components
     - template: /eng/pipelines/common/upload-artifact-step.yml
       parameters:
-        rootFolder: $(managedTestArtifactRootFolderPath)
```
Comment (trylek) [inline eng/pipelines/common/templates/runtimes/build-test-job.yml:150]: Nit - should we redefine <code>managedTestArtifactRootFolderPath</code> rather than open-coding the <code>rootFolder</code> so that we don't get out of sync in other yml scripts?

**Review by trylek — APPROVED** [review]
LGTM, thanks Jeremy for fixing this long-standing debt!

File: src/tests/build.proj [hunk 2]
```diff
@@ -448,14 +448,17 @@
...
       <GroupBuildCmd>$(GroupBuildCmd) "/p:TargetOS=$(TargetOS)"</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) "/p:RuntimeOS=$(RuntimeOS)"</GroupBuildCmd>
       <GroupBuildCmd>$(GroupBuildCmd) "/p:RuntimeFlavor=$(RuntimeFlavor)"</GroupBuildCmd>
       <GroupBuildCmd>$(GroupBuildCmd) "/p:RuntimeVariant=$(RuntimeVariant)"</GroupBuildCmd>
       <GroupBuildCmd>$(GroupBuildCmd) "/p:CLRTestBuildAllTargets=$(CLRTestBuildAllTargets)"</GroupBuildCmd>
       <GroupBuildCmd>$(GroupBuildCmd) "/p:__TestGroupToBuild=$(__TestGroupToBuild)"</GroupBuildCmd>
       <GroupBuildCmd>$(GroupBuildCmd) "/p:__SkipRestorePackages=1"</GroupBuildCmd>
       <GroupBuildCmd>$(GroupBuildCmd) /nodeReuse:false</GroupBuildCmd>
       <GroupBuildCmd>$(GroupBuildCmd) /maxcpucount</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) /bl:$(ArtifactsDir)/log/$(Configuration)/InnerManagedTestBuild.$(__TestGroupToBuild).binlog</GroupBuildCmd>
```
Comment (trylek) [inline src/tests/build.proj:461]: Thanks for putting this back, I think I lost that in my <code>build.proj</code> refactoring.

File: eng/pipelines/common/templates/runtimes/build-test-job.yml (hunk 1 above)
Comment (jkoritzinsky) [reply eng/pipelines/common/templates/runtimes/build-test-job.yml:150]: This is a single-use special case of this path, so I think we want it to be different. The "osGroup" parameter to this job is still not "AnyOS" as it needs to match the libraries build so we can correctly locate the artifacts, but we want to download this artifact into the correct folder in each other build (which is all of the other use cases of `$(managedTestArtifactRootFolderPath).

Comment (trylek) [reply eng/pipelines/common/templates/runtimes/build-test-job.yml:150]: I think that ultimately I would be happier if we could express this in a cleaner way, e.g. by introducing a new parameter "logicalOsGroup" that would be AnyOS in this case and propagating it all the way down to

https://github.com/dotnet/runtime/blob/16300e0aee3392bc604b258db0901d9047e80f7c/eng/pipelines/coreclr/templates/xplat-pipeline-job.yml#L113

but I guess we can offload that to a future cleanup task and possibly present is as an up-for-grabs issue for infra volunteers.

Comment (trylek) [reply eng/pipelines/common/templates/runtimes/build-test-job.yml:150]: In other words, I don't see this as blocking feedback but please consider creating the cleanup issue and adding it to the CoreCLR infrastructure backlog.

Comment (trylek) [reply eng/pipelines/common/templates/runtimes/build-test-job.yml:150]: I actually believe such a task to be very instructive for junior team members eager to ramp up on yml scripts.

File: eng/pipelines/common/templates/runtimes/build-test-job.yml [hunk 3]
```diff
@@ -89,9 +89,13 @@ jobs:
...
         - name: liveRuntimeBuildParams
           value: ${{ format('clr.corelib+libs.ref+libs.native -rc {0} -c {1} -arch {2} -ci', coalesce(parameters.liveRuntimeBuildConfig, parameters.buildConfig), parameters.liveLibrariesBuildConfig, parameters.archType) }}
+        - name: liveRuntimeArtifactsPathArg
+          value: ${{ format('/p:CoreCLROverridePath={0}/artifacts/bin/coreclr/{1}{2}.{3}.{4}', $(Build.SourcesDirectory), parameters.osGroup, parameters.osSubgroup, parameters.archType, coalesce(parameters.liveRuntimeBuildConfig, parameters.buildConfig)) }}
       - ${{ if eq(parameters.runtimeFlavor, 'mono') }}:
         - name: liveRuntimeBuildParams
           value: ${{ format('mono.corelib+libs.ref+libs.native -rc {0} -c {1} -arch {2} -ci', coalesce(parameters.liveRuntimeBuildConfig, parameters.buildConfig), parameters.liveLibrariesBuildConfig, parameters.archType) }}
+        - name: liveRuntimeArtifactsPathArg
+          value: ${{ format('/p:MonoOverridePath={0}/artifacts/bin/mono/{1}{2}.{3}.{4}', $(Build.SourcesDirectory), parameters.osGroup, parameters.osSubgroup, parameters.archType, coalesce(parameters.liveRuntimeBuildConfig, parameters.buildConfig)) }}
```
Comment (safern) [inline eng/pipelines/common/templates/runtimes/build-test-job.yml:98]: Curious and to understand, why is this needed now? 

We didn't use the runtime to build the tests?

Comment (jkoritzinsky) [reply eng/pipelines/common/templates/runtimes/build-test-job.yml:98]: This is needed now because there's no alternative to specify a different path to the CoreCLR/Mono artifacts for System.Private.CoreLib without overriding the whole path (unlike Libraries, which has overrides for various different paths that we can do when we set TargetOS=AnyOS)

Comment (jkoritzinsky) [reply eng/pipelines/common/templates/runtimes/build-test-job.yml:98]: With https://github.com/dotnet/runtime/pull/61754 or some equivalent PR, we can remove this since we won't be referencing System.Private.CoreLib directly.

Comment (safern) [reply eng/pipelines/common/templates/runtimes/build-test-job.yml:98]: Ah, ok, makes sense.

File: src/tests/build.proj [hunk 4]
```diff
@@ -448,14 +446,17 @@
       <GroupBuildCmd>$(GroupBuildCmd) "/p:TargetArchitecture=$(TargetArchitecture)"</GroupBuildCmd>
       <GroupBuildCmd>$(GroupBuildCmd) "/p:Configuration=$(Configuration)"</GroupBuildCmd>
       <GroupBuildCmd>$(GroupBuildCmd) "/p:LibrariesConfiguration=$(LibrariesConfiguration)"</GroupBuildCmd>
+      <GroupBuildCmd>$(GroupBuildCmd) "/p:LibrariesTargetOSConfigurationArchitecture=$(LibrariesTargetOSConfigurationArchitecture)"</GroupBuildCmd>
```
Comment (safern) [inline src/tests/build.proj:449]: We usually refer to this in the libraries world as `BuildSettings` might make it easier to read this property, something like `LibrariesBuildSettings`

**Review by safern — APPROVED** [review]

File: eng/pipelines/common/templates/runtimes/build-test-job.yml (hunk 1 above)
Comment (jkoritzinsky) [reply eng/pipelines/common/templates/runtimes/build-test-job.yml:150]: Tracking in https://github.com/dotnet/runtime/issues/61790

**Comment by jkoritzinsky** [comment]
Timeouts seem to be happing across multiple PRs and occur in legs that don't use this job, so I'm going to merge.
